docs/design/admission_control_limit_range.md

```diff
@@ -35,139 +35,184 @@ Documentation for other releases can be found at
 
 ## Background
 
-This document proposes a system for enforcing min/max limits per resource as part of admission control.
+This document proposes a system for enforcing resource requirements constraints as part of admission control.
 
-## Model Changes
+## Use cases
 
-A new resource, **LimitRange**, is introduced to enumerate min/max limits for a resource type scoped to a
-Kubernetes namespace.
+1. Ability to enumerate resource requirement constraints per namespace
+2. Ability to enumerate min/max resource constraints for a pod
+3. Ability to enumerate min/max resource constraints for a container
+4. Ability to specify default resource limits for a container
+5. Ability to specify default resource requests for a container
+6. Ability to enforce a ratio between request and limit for a resource.
+
+## Data Model
+
+The **LimitRange** resource is scoped to a **Namespace**.
+
+### Type
 
 ```go
+// A type of object that is limited
+type LimitType string
+
 const (
   // Limit that applies to all pods in a namespace
-  LimitTypePod string = "Pod"
+  LimitTypePod LimitType = "Pod"
   // Limit that applies to all containers in a namespace
-  LimitTypeContainer string = "Container"
+  LimitTypeContainer LimitType = "Container"
 )
 
 // LimitRangeItem defines a min/max usage limit for any resource that matches on kind
 type LimitRangeItem struct {
   // Type of resource that this limit applies to
-  Type string `json:"type,omitempty"`
+  Type LimitType `json:"type,omitempty" description:"type of resource that this limit applies to"`
   // Max usage constraints on this kind by resource name
-  Max ResourceList `json:"max,omitempty"`
+  Max ResourceList `json:"max,omitempty" description:"max usage constraints on this kind by resource name"`
   // Min usage constraints on this kind by resource name
-  Min ResourceList `json:"min,omitempty"`
-  // Default usage constraints on this kind by resource name
-  Default ResourceList `json:"default,omitempty"`
+  Min ResourceList `json:"min,omitempty" description:"min usage constraints on this kind by resource name"`
+  // Default resource limits on this kind by resource name
+  Default ResourceList `json:"default,omitempty" description:"default resource limits values on this kind by resource name if omitted"`
+  // DefaultRequests resource requests on this kind by resource name
+  DefaultRequests ResourceList `json:"defaultRequests,omitempty" description:"default resource requests values on this kind by resource name if omitted"`
+  // LimitRequestRatio is the ratio of limit over request that is the maximum allowed burst for the named resource
+  LimitRequestRatio ResourceList `json:"limitRequestRatio,omitempty" description:"the ratio of limit over request that is the maximum allowed burst for the named resource"`
```

Comment:
Minor correction: Unsetting limit won't help. If there is a LimitRequestRatio, then limit will have to be set to a value less or equal to LimitRequestRatio*request if request is set.

I thought about making infinite limit imply best effort, but then request would have an inconsistent meaning, so I don't think that's viable.